### PR TITLE
Feature/perf

### PR DESCRIPTION
**Pull request by jasonwyatt**
Goal for this iteration is to mostly reduce overall object allocations.

### CODE REVIEW

File: src/main/java/redis/clients/jedis/XReadArgs.java [hunk 1]
```diff
@@ -2,40 +2,96 @@
...
+    private XReadArgs(List<String> streamsAndOffsets, int count, boolean block, long blockDuration) {
+        this.streamsAndOffsets = streamsAndOffsets;
         this.count = count;
         this.block = block;
         this.blockDuration = blockDuration;
     }
 
+    public void setStreamsAndOffsets(List<String> streamsAndOffsets) {
+        this.streamsAndOffsets.clear();
+        if (streamsAndOffsets.size() < 50) {
```
Comment (markrebhan) [inline src/main/java/redis/clients/jedis/XReadArgs.java]: Is there any need to arbitrarily  call add all after size is larger than 50? Might as well just call add no matter the size unless I am missing something...

File: src/main/java/redis/clients/jedis/XReadArgs.java [hunk 2]
```diff
@@ -2,40 +2,96 @@
...
+
     public static Builder builder() {
         return new Builder();
     }
 
     public static class Builder {
 
-        private final List<String> streams = new LinkedList<>();
-        private final List<String> offsets = new LinkedList<>();
+        private final List<String> streams = new ArrayList<>();
```
Comment (markrebhan) [inline src/main/java/redis/clients/jedis/XReadArgs.java:96]: Not sure why this is not a linked list...

**Review by markrebhan — CHANGES_REQUESTED** [review]

File: src/main/java/redis/clients/jedis/XReadArgs.java (hunk 1 above)
Comment (jasonwyatt) [reply src/main/java/redis/clients/jedis/XReadArgs.java]: just wanted to take advantage of System.arrayCopy in situations where the list is super long.  Trading an allocation for performance when using `addAll`

File: src/main/java/redis/clients/jedis/XReadArgs.java (hunk 2 above)
Comment (jasonwyatt) [reply src/main/java/redis/clients/jedis/XReadArgs.java:96]: Linked lists create an object for every item you add to them. Array lists don't

**Review by markrebhan — APPROVED** [review]

**Pull request merged**